Description: Fix jobs with multiple files or multiple formats
Author: Tim Waugh <twaugh@redhat.com>
Bug-Redhat: https://bugzilla.redhat.com/show_bug.cgi?id=972242
Last-Updated: 2013-11-28
--- a/backend/ipp.c
+++ b/backend/ipp.c
@@ -1612,7 +1612,10 @@
 	ippAddBoolean(request, IPP_TAG_OPERATION, "last-document",
         	      (i + 1) >= num_files);
 
-	if (document_format)
+	if (num_files > 1)
+	  ippAddString(request, IPP_TAG_OPERATION, IPP_TAG_MIMETYPE,
+		       "document-format", NULL, "application/octet-stream");
+	else if (document_format)
 	  ippAddString(request, IPP_TAG_OPERATION, IPP_TAG_MIMETYPE,
 		       "document-format", NULL, document_format);
 
